coniferest/onnx/operator_converters/aadforest.py

```diff
@@ -0,0 +1,85 @@
+from onnxconverter_common.registration import register_converter
+
+
+def get_default_attribute_pairs():
+    attrs = {}
+    attrs['post_transform'] = 'NONE'
+    attrs['n_targets'] = 1
+    attrs['nodes_treeids'] = []
+    attrs['nodes_nodeids'] = []
+    attrs['nodes_featureids'] = []
+    attrs['nodes_modes'] = []
+    attrs['nodes_values'] = []
+    attrs['nodes_truenodeids'] = []
+    attrs['nodes_falsenodeids'] = []
+    attrs['target_treeids'] = []
+    attrs['target_nodeids'] = []
+    attrs['target_ids'] = []
+    attrs['target_weights'] = []
+    return attrs
+
+def get_leaf_weight(tree_id, node_id, parents, model):
+    evaluator   = model.evaluator
+    selector_id = evaluator.indices[tree_id] + node_id
+    selector    = evaluator.selectors[selector_id]
+    value       = selector["value"]
+    # Generally, selector["leaf"] should be -1 for leafs,
+    # but for AADForest it stores related weight index.
+    leaf_id     = selector["left"]
+    weight      = evaluator.weights[leaf_id]
+
+    return weight * value
+
+def add_node(attr_pairs, tree_id, node_id, feature_id, mode, value, true_child_id, false_child_id):
+    attr_pairs['nodes_treeids'].append(tree_id)
+    attr_pairs['nodes_nodeids'].append(node_id)
+    attr_pairs['nodes_featureids'].append(feature_id)
+    attr_pairs['nodes_modes'].append(mode)
+    attr_pairs['nodes_values'].append(value)
+    attr_pairs['nodes_truenodeids'].append(true_child_id)
+    attr_pairs['nodes_falsenodeids'].append(false_child_id)
+
+def add_leaf(attr_pairs, tree_id, node_id, mode, weight):
+    add_node(attr_pairs, tree_id, node_id, 0, mode, 0.0, 0, 0)
+
+    attr_pairs['target_treeids'].append(tree_id)
+    attr_pairs['target_nodeids'].append(node_id)
+    attr_pairs['target_ids'].append(0)
+    attr_pairs['target_weights'].append(weight)
+
+def add_tree_to_attribute_pairs(attr_pairs, tree, tree_id, model):
+    parents = {}
+
+    for i in range(tree.node_count):
+        node_id = i
+
+        if tree.children_left[i] > i or tree.children_right[i] > i:
```

Comment:
I don't understand this, could you please leave a comment why for the leaf children values are less than `i`?

Reply:
As far as I remember, I steal this code from skl2onnx :-)

Reply:
@matwey do we have the same tree implementation as there? I'm ok about it we now what we are doing...